File: day_5/part_1.py
def get_inp(inp):
    ranges = []
    ids = []
    with open(inp, "r") as file:
        lines = file.readlines()
        for line in lines:
            line = line.strip()
            if not len(line):
                continue
            if "-" in line:
                ranges.append(list(map(lambda x: int(x), line.split("-"))))
            else:
                ids.append(int(line))
    return ranges, ids


def check(num, ranges):
    for x, y in ranges:
        if num < x:
            return False
        if num >= x and num <= y:
            return True
    return False


def solve(inp):
    ranges, ids = get_inp(inp)
    ranges.sort()
    return len(list(filter(lambda x: x, list(map(lambda x: check(x, ranges), ids)))))


# print(solve("example.txt"))
print(solve("input.txt"))

File: day_5/part1.js
const fs = require("fs");
const FILEPATH = "./input.txt";
const filecontent = fs.readFileSync(FILEPATH, "utf-8").split("\n");

const check = (num, ranges) => {
  for (let [x, y] of ranges) {
    if (num < x) return false;
    if (num >= x && num <= y) return true;
  }
  return false;
};

const solve = (inp) => {
  let [ranges, ids] = [
    inp.slice(0, inp.indexOf("")),
    inp.slice(inp.indexOf("") + 1),
  ];
  ranges = ranges
    .map((x) => x.split("-").map((x) => +x))
    .sort((x, y) => x[0] - y[0]);
  ids = ids.map((x) => +x);

  return ids.reduce((acum, cur) => (check(cur, ranges) ? 1 + acum : acum), 0);
};

console.log(solve(filecontent));
